### ISSUE TITLE
Chart not updated (regression)

### ISSUE THREAD

**Issue by panuhorsmalahti** [ngyewch/angular-rickshaw#11]
For some reason angular rickshaw 0.5.0 works well, but in angular rickshaw 0.6.0 the chart is only updated once. My debugging so far has found that the update function is called in the directive, but for some reason rickshaw itself doesn't get the new series.

**Comment by ngyewch** [comment]
Hmm, that's strange. The last part of the demo tests angular-rickshaw against changes in series data, is it similar to what you're doing?

Maybe if you have a code snippet?

**Comment by panuhorsmalahti** [comment]
I've created a wrapper directive on top of angular-rickshaw. I'm refreshing the data by reconstructing the rickshaw-series object (although the only truly changing values are in the .data arrays.

My template is simply
```
rickshaw(rickshaw-options="options", rickshaw-series="series", rickshaw-features="features")
```
And I'm updating the series with
```
scope.series = newSeriesObject();
```

It's not a basic scope.$apply issue, as the apply does nothing (and is not necessary according to my debugging).

**Comment by ngyewch** [comment]
Could you try it without your wrapper directive? The problem may be caused by your wrapper.
The way the directive detects changes in the data did change from 0.5.0 to 0.6.0.

By default, it only watches for changes in the first series.

**Comment by panuhorsmalahti** [comment]
I'm now fairly certain the bug is caused by this:
https://github.com/ngyewch/angular-rickshaw/blob/master/rickshaw.js#L139

The watchers are created to watch particular arrays when the directive is created. However, when the arrays are re-created or changed, the watchers are not updated. Therefore, they will continue to watch the old arrays which never change.

I implemented a workaround in my directive which reuses the same arrays which solves the issue.

One fix would be to revert the 'efficient' watcher change. I don't think it's majorly more efficient as the series objects are not that much larger than the data arrays.

**Comment by panuhorsmalahti** [comment]
Another solution would be to create a "updateWatchers" type of function, which would recreate the watchers according to the new series object.

**Comment by ngyewch** [comment]
Yeah, I think I'll revert it back to watch the entire object rather than just the series array.

**Comment by ngyewch** [comment]
I've released 0.10.0 to address this issue. Hope this works for you.

**Comment by panuhorsmalahti** [comment]
Thanks, 0.10.0 fixed the issue! :+1:
